src/main/java/org/folio/search/controller/ApiExceptionHandler.java

```diff
@@ -0,0 +1,153 @@
+   */
+  @ExceptionHandler(SearchOperationException.class)
+  public ResponseEntity<ErrorResponse> handleSearchServiceException(SearchOperationException exception) {
+    log.debug("Handling exception", exception);
+
+    var cause = exception.getCause();
+    return cause instanceof ElasticsearchException
+      ? handleElasticsearchException((ElasticsearchException) cause)
+      : buildResponseEntity(exception, INTERNAL_SERVER_ERROR, exception.getErrorCode());
+  }
+
+  /**
+   * Catches and handles all exceptions of type {@link MethodArgumentNotValidException}.
+   *
+   * @param e {@link MethodArgumentNotValidException} to process
+   * @return {@link ResponseEntity} with {@link ErrorResponse} body
+   */
+  @ExceptionHandler(MethodArgumentNotValidException.class)
+  public ResponseEntity<ErrorResponse> handleMethodArgumentNotValidException(MethodArgumentNotValidException e) {
+    var validationErrors = Optional.of(e.getBindingResult()).map(Errors::getAllErrors).orElse(emptyList());
+    var errorResponse = new ErrorResponse();
+    validationErrors.forEach(error ->
+      errorResponse.addErrorsItem(new Error()
+        .message(error.getDefaultMessage())
+        .code(ErrorCode.VALIDATION_ERROR.getDescription())
+        .type(MethodArgumentNotValidException.class.getSimpleName())
+        .addParametersItem(new Parameter()
+          .key(((FieldError) error).getField())
+          .value(String.valueOf(((FieldError) error).getRejectedValue())))));
+    errorResponse.totalRecords(errorResponse.getErrors().size());
+
+    return buildResponseEntity(errorResponse, BAD_REQUEST);
+  }
+
+  /**
+   * Catches and handles all exceptions of type {@link ConstraintViolationException}.
+   *
+   * @param exception {@link ConstraintViolationException} to process
+   * @return {@link ResponseEntity} with {@link ErrorResponse} body
+   */
+  @ExceptionHandler(ConstraintViolationException.class)
+  public ResponseEntity<ErrorResponse> handleConstraintViolation(ConstraintViolationException exception) {
+    log.debug("Handling exception", exception);
+    var errorResponse = new ErrorResponse();
+    exception.getConstraintViolations().forEach(constraintViolation ->
+      errorResponse.addErrorsItem(new Error()
+        .message(constraintViolation.getMessage())
+        .code(ErrorCode.VALIDATION_ERROR.getDescription())
+        .type(ConstraintViolationException.class.getSimpleName())));
+    errorResponse.totalRecords(errorResponse.getErrors().size());
+
+    return buildResponseEntity(errorResponse, BAD_REQUEST);
+  }
+
+  /**
+   * Handles all uncaught exceptions.
+   *
+   * @param exception {@link Exception} object
+   * @return {@link ResponseEntity} with {@link ErrorResponse} body.
+   */
+  @ExceptionHandler(Exception.class)
+  public ResponseEntity<ErrorResponse> handleAllOtherExceptions(Exception exception) {
+    log.warn("Handling exception [type: {}]", exception.getClass().getSimpleName(), exception);
```

Comment:
For consistence, maybe just `log.warn("Handling exception", exception);`. The stack trace will contain exception class name.